doc/tutorials/highlevel-bindings.md

```diff
@@ -54,6 +54,29 @@ sufficient to write (or generate) a one-to-one mirror of the C API and use that
 The generated code should still be understandable. So if writing a more idiomatic API on top of the
 direct mapping, significantly simplifies the generated code (and maybe also the template), you should write such an API.
 
+### Bindings for the `lowlevel` Library
+
+If you decide on mapping the functionality of kdb.h 1:1 it is pretty straight-forward.
+
+Your programming language of choice must provide a way to call into C code, e.g.: 
+* [cgo](https://golang.org/cmd/cgo/) for the Go programming language
+* [jni](https://docs.oracle.com/javase/8/docs/technotes/guides/jni/) for JAVA
+* [n-api](https://nodejs.org/api/n-api.html) for javascript
+
+Make sure you have a good understanding of these libraries because some of them can have unexpected downsides.
+CGO for example cannot call variadic C functions because in C the amount of parameters has to be known at compile-time. This is not the case with Go
```

Comment:
Give variadic C functions an own section as this is always something problematic.

@PhilippGackstatter how did you solve it?

Reply:
Rust allows calling variadic functions, but not declaring them. I described this as a possible use case for the raw bindings in Rust [here](https://github.com/ElektraInitiative/libelektra/tree/master/src/bindings/rust#raw-bindings).
A more general solution, applicable to many languages is to us the builder pattern. With this, it's easy to set multiple meta values in one go, just like with the variadic `keyNew`. For example, in the Rust binding.

```rust
let key: StringKey = KeyBuilder::new("user/test/key")?
            .value("value")
            .meta("metaname", "metavalue")?
            .meta("OWNER", "me")?
            .build();
```

Reply:
Actually, it would be good if we mention this in the binding tutorial.